### PR TITLE
Action Hub split inputs

### PR DESCRIPTION
**Pull request by johnhwhite**
* allow separate inputs
* breaking change: remove `data` input
* use SkyHref
* support navigation extras for needs attention items
* show message when there are no needs attention items

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Review by Blackbaud-PaulCrowder — DISMISSED** [review]

File: src/assets/locales/resources_en_US.json [hunk 1]
```diff
@@ -10,5 +10,9 @@
   "sky_action_hub_needs_attention": {
     "_description": "Heading for needs attention items",
     "message": "Needs Attention"
+  },
+  "sky_action_hub_needs_attention_empty": {
+    "_description": "Display when the needs attention list is empty",
+    "message": "The needs attention list is currently empty"
```
Comment (blackbaud-paulah) [inline src/assets/locales/resources_en_US.json]:
```suggestion
    "message": "No issues currently need attention"
```

File: src/app/public/modules/action-hub/action-hub.component.ts [hunk 2]
```diff
@@ -15,47 +14,28 @@ import { SkyRecentLink } from './types/recent-link';
...
-        value.relatedLinks
-      );
-      this.title = value.title;
-    }
-  }
-
-  public needsAttention: SkyActionHubNeedsAttention[];
+  public needsAttention: SkyActionHubNeedsAttention[] | 'loading' = [];
 
+  @Input()
```
Comment (blackbaud-johnly) [inline src/app/public/modules/action-hub/action-hub.component.ts:26]: Don't we need JDocs comments on these public inputs?

File: src/app/public/modules/action-hub/action-hub.component.ts [hunk 3]
```diff
@@ -16,46 +15,42 @@ import { SkyRecentLink } from './types/recent-link';
...
 
+  /**
+   * Links back to a parent page if needed.
+   */
+  @Input()
   public parentLink: SkyPageLink;
 
-  public recentLinks: SkyRecentLink[];
+  /**
+   * Provides a list of recently access links, or "loading" to show a waiting indicator.
```
Comment (blackbaud-paulah) [inline src/app/public/modules/action-hub/action-hub.component.ts]:
```suggestion
   * Provides a list of recently accessed links, or "loading" to show a waiting indicator.
```

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

File: src/app/public/modules/action-hub/action-hub.component.ts [hunk 4]
```diff
@@ -16,46 +15,42 @@ import { SkyRecentLink } from './types/recent-link';
 })
 export class SkyActionHubComponent {
   /**
-   * Passes a SkyActionHubData object to build the action hub. The page loads until
-   * `[data]` has a value.
-   *
-   * @param value
+   * Provides a list of items that need attention, or "loading" to show a waiting indicator.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/action-hub/action-hub.component.ts]:
```suggestion
   * Provides a list of actions that users must perform based on business requirements or best practices, or `'loading'` to display a wait indicator.
```

File: src/app/public/modules/action-hub/action-hub.component.ts [hunk 5]
```diff
@@ -16,46 +15,42 @@ import { SkyRecentLink } from './types/recent-link';
...
-      );
-      this.title = value.title;
-    }
-  }
-
-  public needsAttention: SkyActionHubNeedsAttention[];
+  public needsAttention: SkyActionHubNeedsAttention[] | 'loading' = [];
 
+  /**
+   * Links back to a parent page if needed.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/action-hub/action-hub.component.ts]:
```suggestion
   * Links back to a parent page.
```

File: src/app/public/modules/action-hub/action-hub.component.ts [hunk 6]
```diff
@@ -16,46 +15,42 @@ import { SkyRecentLink } from './types/recent-link';
...
-  public recentLinks: SkyRecentLink[];
+  /**
+   * Provides a list of recently access links, or "loading" to show a waiting indicator.
+   */
+  @Input()
+  public recentLinks: SkyRecentLink[] | 'loading' = [];
 
-  public relatedLinks: SkyPageLink[];
+  /**
+   * Provides a list of related links, or "loading" to show a waiting indicator.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/action-hub/action-hub.component.ts]:
```suggestion
   * Provides a list of related links, or `'loading'` to display a wait indicator.
```

File: src/app/public/modules/action-hub/action-hub.component.ts [hunk 7]
```diff
@@ -16,46 +15,42 @@ import { SkyRecentLink } from './types/recent-link';
...
+   */
+  @Input()
+  public relatedLinks: SkyPageLink[] | 'loading' = [];
 
+  /**
+   * Sets the H1 title for the page.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/action-hub/action-hub.component.ts]:
```suggestion
   * Specifies the page title.
```

File: src/app/public/modules/action-hub/action-hub.component.ts (hunk 3 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/action-hub/action-hub.component.ts]:
```suggestion
   * Provides a list of recently accessed links, or `'loading'` to display a wait indicator.
```

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs](https://developer.blackbaud.com/skyux-pages?_sp.spa=pr-action-hub-split-inputs)

**Review by blackbaud-paulah — APPROVED** [review]

**Review by blackbaud-johnly — APPROVED** [review]

**Pull request merged**